src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx

```diff
@@ -1,18 +1,23 @@
 import React from 'react';
+import { useRouter } from 'next/router';
 import { FaRegClock } from 'react-icons/fa6';
 import {
     LatestProgressUpdateCardPresentationProps,
     ProgressUpdateDataToShow,
 } from './progressUpdateCard.types';
 import Tooltip from '@/components/common/Tooltip/Tooltip';
 import styles from './latest-progress-update-card.module.scss';
+import { USER_MANAGEMENT_URL } from '@/constants/url';
 
 export default function LatestProgressUpdateCardPresentation({
+    username,
     dataToShowState,
     tooltipText,
     onMoreOrLessButtonClick,
     dateInAgoFormat,
 }: LatestProgressUpdateCardPresentationProps) {
+    const router = useRouter();
+    const isDevMode = router.query.dev === 'true';
```

Comment:
Does component needs to be aware of the route?

Reply:
These changes are currently behind  feature flag, so to retrieve the dev query parameter, we need access to the route. Once the feature flag is removed in the future, this setup can be eliminated